ddtrace/contrib/flask/patch.py

```diff
@@ -109,23 +109,28 @@ def _wrapped_start_response(self, start_response, ctx, status_code, headers, exc
         core.dispatch("flask.start_response.pre", (flask.request, ctx, config.flask, status_code, headers))
         if not core.get_item(HTTP_REQUEST_BLOCKED):
             headers_from_context = ""
-            result = core.dispatch_with_results("flask.start_response", ("Flask",)).waf
-            if result:
-                headers_from_context = result.value
+            result_waf = core.dispatch_with_results("flask.start_response", ("Flask",)).waf
+            if result_waf:
+                headers_from_context = result_waf.value
             if core.get_item(HTTP_REQUEST_BLOCKED):
                 # response code must be set here, or it will be too late
-                block_config = core.get_item(HTTP_REQUEST_BLOCKED)
-                desired_type = block_config.get("type", "auto")
-                status = block_config.get("status_code", 403)
-                if desired_type == "none":
-                    response_headers = []
+                result_content = core.dispatch_with_results("flask.block.request.content", ()).block_requested
```

Comment:
Instead of adding new `dispatch_with_results` calls, I'd recommend having the event handler put `result_content` into the current context to make it available to this function. This is because `dispatch_with_results` might change before too long, or be removed completely.

Reply:
We already using dispatch_with_results in all frameworks (25 occurrences , most of them in contrib). 
Thank for the information, I'll remove all of them when it will be needed in a subsequent PR.